src/lua/thread.rs

```diff
@@ -152,10 +152,24 @@ pub fn init() {
         info!("Skipping config search");
     }
 
+    // Only ready after loading libs
+    *RUNNING.write().expect(ERR_LOCK_RUNNING) = true;
+    debug!("Entering main loop...");
+    let _lua_handle = thread::Builder::new()
+        .name("Lua thread".to_string())
+        .spawn(move || { main_loop(receiver, &mut lua) });
+    // Immediately update all the values that the init file set
+    send(LuaQuery::UpdateRegistryFromCache)
+        .expect("Could not update registry from cache");
+}
+
+pub fn on_compositor_ready() {
+    info!("Running lua on_init()");
     // Call the special init hook function that we read from the init file
-    if let Err(error) = lua.execute::<()>(INIT_LUA_FUNC) {
-        error!("Lua init callback returned an error: {:?}", error);
-    }
+    send(LuaQuery::Execute(INIT_LUA_FUNC.to_owned()))
+        .err()
+        .map(|error| { error!("Lua init callback returned an error: {:?}", error); error });
+    
     // Re-tile the layout tree, to make any changes appear immediantly.
     if let Ok(mut tree) = lock_tree() {
```

Comment:
This re-tile needs to be in `init`, not `on_compositor_ready`. If you restart the Lua thread, this method is never called (which is the correct logic). However, that means if a value that affects the layout is updated (e.g, you change the gap size or the colour of the borders for example) then it won't take effect immediately.

So please move this into `lua::thread::init`, right after you send the request to update the registry from the cache.